src/OrbitGl/TimeGraphLayout.h

```diff
@@ -16,6 +16,8 @@ class TimeGraphLayout {
   float GetTextCoresHeight() const { return core_height_ * scale_; }
   float GetThreadStateTrackHeight() const { return thread_state_track_height_ * scale_; }
   float GetEventTrackHeight() const { return event_track_height_ * scale_; }
+  float GetAllThreadsEventTrackHeight() const { return all_threads_event_track_height_ * scale_; }
+  float GetEventTrackHeightFromTid(int32_t tid) const;
```

Comment:
I don't like to have both GetEventTrackHeight and GetEventTrackHeightFromTid public in the Layout. Could be error prone.

I'd prefer to have one of this possibilities:
1- Making both GetEventTrackHeight and GetAllThreadsEventTrackHeight private, so we only can use the new public function to get the Height. (better for me)
2- Only include the new GetAllThreadsEventTrackHeight and move the logic of choosing to ThreadTrack. (which I'm not sure that I like it).

Reply:
Yeah I had # 2 originally and didn't like it. # 1 sounds good though. The only weird thing is GraphTrack.cpp - it looks like it's tid i also set to all threads and I don't want to make that track any larger. My solution was to put 0 as the tid, but the reasoning is non-obvious. It might be better to have both GetEventTrackHeightFromTid and GetEventTrackHeight if tid doesn't matter. What do you think?

Reply:
Now that I'm seeing the screenshot. Don't we have the same problem with AsyncTracks? At least that EventTrack seems to be also large but I really don't remember how it was before.

The fact of sharing the same tid from all threads and the GraphTracks seems to be an issue we should solve. IMO, we should have an unique tid for AllThreadTrack. I supposed that it was unique and not shared (and "no tid", as in GraphTrack, was 0), now I'm confused. 

1- Do you know where we are setting that constant for GraphTrack?
2- Should we create a constant for "no thread id" in ThreadConstant and use that?

What do you think @pierricgimmig ?

Reply:
Yep we are using -1 as the "invalid" thread id, but this clashes with `kAllProcessThreadsTid` in ThreadConstants.h. I agree with you Pablo, we should add a `kInvalidThreadId` constant in there and use it as the default value for tracks. Right now, the -1 is hardcoded in the `Track` constructor.

Reply:
I set kInvalidThreadId to 0, but if that's problematic or clashes with something else, I can set it to -4.

Reply:
PID 0 is possible on Linux, "There are two tasks with specially distinguished process IDs: swapper or sched has process ID 0 and is responsible for paging", https://en.wikipedia.org/wiki/Process_identifier. So TID 0 might as well be possible in the case where we receive data from `swapper` or `sched`. I think we should use a negative number, -4 seems fine to me!

Reply:
done :)